src/main/java/com/exactpro/th2/crawler/Crawler.java

```diff
@@ -165,57 +183,84 @@ public Duration process() throws IOException, UnexpectedDataProcessorException {
             intervalStartForProcessor(dataProcessor, interval, state);
 
             Continuation continuation = state == null ? null : typeStrategy.continuationFromState(state);
-            DataParameters parameters = new DataParameters(info, crawlerId, sessionAliases);
-            CrawlerData<Continuation> data;
-            Report<Continuation> sendingReport = null;
-            long processedEvents = 0L;
-            Timestamp startTime = toTimestamp(interval.getStartTime());
-            Timestamp endTime = toTimestamp(interval.getEndTime());
+
+            Collection<String> newAliases = null;
+
+            int counter = 0;
+
             do {
-                LOGGER.trace("Requesting data for interval");
-                data = requestData(startTime, endTime, continuation, parameters);
-                continuation = data.getContinuation();
-                var currentData = data;
-                long remaining = sendingReport == null ? 0 : sendingReport.getRemainingData();
-                sendingReport = currentData.getHasData()
-                        ? processData(currentInt, parameters, currentData)
-                        : requireNonNullElse(sendingReport, Report.empty());
-
-                if (currentData.getHasData()) {
-                    metrics.updateProcessedData(crawlerType, currentData.size());
+                counter++;
+
+                DataParameters parameters;
+
+                // At first, we need to process messages for all session aliases we have.
+                // If we have a regexp, we need to process new aliases while we keep getting them.
+                if (counter == 1) {
+                    parameters = new DataParameters(info, crawlerId, sessionAliases);
+                    LOGGER.debug("Crawler is going to process messages with {} aliases: {}. Interval from {} to {}",
+                            sessionAliases.size(), String.join(", ", sessionAliases), interval.getStartTime(), interval.getEndTime());
+                } else {
+                    parameters = new DataParameters(info, crawlerId, newAliases);
+                    sessionAliases.addAll(newAliases);
+                    LOGGER.debug("Crawler got {} extra aliases: {}. They will be processed on interval from {} to {}",
+                            newAliases.size(), String.join(", ", newAliases), interval.getStartTime(), interval.getEndTime());
```

Comment:
Maybe we should execute the first part outside the loop and move the second part to the end of the loop?